### PR TITLE
[ENH] input checks for `BaseBenchmark`, allow `add_estimator` to accept multiple estimators

### PR DESCRIPTION
**Pull request by hazrulakmal**
See #4873 

The current ForecastingBenchmark interface requires manually looping over multiple estimators to benchmark multiple models, as the add_estimator interface only accepts one estimator per call. This PR provides an option for users to add multiple models with a single add_estimator call while maintaining the existing interface. The interface for multiple models looks as follows:

```python
from sktime.benchmarking.forecasting import ForecastingBenchmark
from sktime.forecasting.naive import NaiveForecaster
from sktime.forecasting.arima import AutoARIMA

benchmark = ForecastingBenchmark()

# for auto naming
models = [NaiveForecaster(), AutoARIMA()]
benchmark.add_estimator(models)

# for custom naming
models ={"Arima-v1":AutoARIMA(),
        "Naive-v1":NaiveForecaster()}
benchmark.add_estimator(models)

# existing interface still works
benchmark.add_estimator(NaiveForecaster())
benchmark.add_estimator(AutoARIMA())
```

### Questions
1. are there any other preferences in terms of the interface?
2. In sktime, are there any best practices regarding where to place util functions within the module? Should I create another folder called *_utils.py* to place any benchmarking util functions, or is the existing solution sufficient?

FYI @fkiraly @marrov

### CODE REVIEW

**Comment by hazrulakmal** [comment]
> We should also test for it though?

Done add test for checking registered multiple estimators here [75d4d93](https://github.com/sktime/sktime/pull/4877/commits/75d4d93d0c5b729bfd787726e9f54f20dab1bafc)

**Comment by hazrulakmal** [comment]
Hi @fkiraly @benHeid @yarnabrina @achieveordie, can anyone retrigger the CI workflow or merge this PR if there's no objection? Thanks!

the failure is unrelated to this PR.

File: sktime/benchmarking/benchmarks.py
Comment (yarnabrina) [inline sktime/benchmarking/benchmarks.py:13]: Real nitpicky non-blocker suggestion: can you rename it to `coerce` from `coer` (in all relevant places)? Personally as a user/developer I'll appreciate seeing 2 more characters to readily understand what the function is supposed to do.

File: sktime/benchmarking/benchmarks.py
Comment (yarnabrina) [inline sktime/benchmarking/benchmarks.py:29]: Is it worth checking that all values are `BaseEstimator` objects, as per docstring?

File: sktime/benchmarking/benchmarks.py
Comment (yarnabrina) [inline sktime/benchmarking/benchmarks.py:31]: Same as above.

Side comment: Pydantic helps to validate these things on it's own just by adding decorators, but I don't think we're ready to start using it. May be this can be discussed in one of the Friday calls.

File: sktime/benchmarking/benchmarks.py
Comment (yarnabrina) [inline sktime/benchmarking/benchmarks.py:42]:
```suggestion
            "estimator must be of a type a dict, list or "
            f"BaseEstimator object but received {type(estimators)}"
```

File: sktime/benchmarking/benchmarks.py
Comment (yarnabrina) [inline sktime/benchmarking/benchmarks.py:40]:
```suggestion
        raise TypeError(
```

**Review by yarnabrina — CHANGES_REQUESTED** [review]

File: sktime/benchmarking/benchmarks.py
Comment (hazrulakmal) [reply sktime/benchmarking/benchmarks.py]: ofcourse! you are right!

File: sktime/benchmarking/benchmarks.py
Comment (hazrulakmal) [reply sktime/benchmarking/benchmarks.py:66]: I added a validation function to check for the estimators object. Pls kindly review them again. 

I think most of the validations in sktime is done natively. adding pydantic means we are proposing adding pydantic as a dev dependency, right? Most likely need to go through meeting(s) with other core developers if im not mistaken.

Comment (yarnabrina) [reply sktime/benchmarking/benchmarks.py:55]: Agreed. I love to use pydantic, but for sktime it'd be a super big effort for sure. That's why made it a side note.

**Review by yarnabrina — APPROVED** [review]
Looks good to me. Added 2 minor non-blocking comments.

File: sktime/benchmarking/benchmarks.py
Comment (hazrulakmal) [reply sktime/benchmarking/benchmarks.py:33]: a bit difficult to do that in this form. How about providing the users with a better error message like "One or many estimator(s) is not an initialised BaseEstimator object(s). Please instantiate the estimator(s) first."

File: sktime/benchmarking/benchmarks.py
Comment (hazrulakmal) [reply sktime/benchmarking/benchmarks.py:31]: yup it does. Thanks for pointing it out!

**Review by fkiraly — APPROVED** [review]
Thanks, request re tests has been addressed.

**Pull request merged**
